lib/standard/text/flat.nit

```diff
@@ -144,29 +183,38 @@ class FlatString
 	#              String Specific Methods           #
 	##################################################
 
-	# Low-level creation of a new string with given data.
+	# Low-level creation of a new string with minimal data.
 	#
 	# `items` will be used as is, without copy, to retrieve the characters of the string.
 	# Aliasing issues is the responsibility of the caller.
-	private init with_infos(items: NativeString, length: Int, from: Int, to: Int)
+	private init with_infos(items: NativeString, bytelen, from, to: Int)
 	do
 		self.items = items
-		self.length = length
-		index_from = from
-		index_to = to
+		self.bytelen = bytelen
+		first_byte = from
+		last_byte = to
 	end
 
-	redef fun to_cstring
+	# Low-level creation of a new string with all the data.
+	#
+	# `items` will be used as is, without copy, to retrieve the characters of the string.
+	# Aliasing issues is the responsibility of the caller.
+	private init full(items: NativeString, bytelen, from, to, length: Int)
 	do
-		if real_items != null then
-			return real_items.as(not null)
-		else
-			var newItems = new NativeString(length + 1)
-			self.items.copy_to(newItems, length, index_from, 0)
-			newItems[length] = 0u8
-			self.real_items = newItems
-			return newItems
-		end
+		self.items = items
+		self.length = length
+		self.bytelen = bytelen
+		first_byte = from
+		last_byte = to
+	end
+
+	redef fun to_cstring do
+		if real_items != null then return real_items.as(not null)
+		var new_items = new NativeString(bytelen + 1)
+		self.items.copy_to(new_items, bytelen, first_byte, 0)
+		new_items[bytelen] = 0u8
+		self.real_items = new_items
```

Comment:
Why is this `real_items` prefixed with `self` where the other are not? It's funny.